Code/Framework/AzToolsFramework/AzToolsFramework/Slice/SliceDependencyBrowserComponent.cpp

```diff
@@ -222,33 +222,53 @@ namespace AzToolsFramework
         }
 
         bool succeeded = m_databaseConnection->QueryDependsOnSourceBySourceDependency(
-            sourceUuid, "%.slice",
+            sourceUuid,
             AzToolsFramework::AssetDatabase::SourceFileDependencyEntry::DEP_SourceOrJob,
             [&](AssetDatabase::SourceFileDependencyEntry& entry)
-        {
-            found = true;
-            dependencies.push_back(AZStd::move(entry.m_dependsOnSource));
-            return true;
-        });
+            {
+                AZStd::string dependencyName = entry.m_dependsOnSource.GetPath();
+
+                if (entry.m_dependsOnSource.IsUuid())
+                {
+                    AzToolsFramework::AssetDatabase::SourceDatabaseEntry sourceEntry;
+                    m_databaseConnection->QuerySourceBySourceGuid(
+                        entry.m_dependsOnSource.GetUuid(),
+                        [&dependencyName](AzToolsFramework::AssetDatabase::SourceDatabaseEntry& entry)
+                        {
+                            dependencyName = entry.m_sourceName;
+                            return false;
+                        });
+                }
+
+                if (!dependencyName.ends_with(".slice"))
+                {
+                    // Filter out non-slice files
+                    return true;
+                }
+
+                found = true;
+                dependencies.push_back(AZStd::move(dependencyName));
+                return true;
+            });
         return found && succeeded;
     }
 
     bool SliceDependencyBrowserComponent::GetSliceDependendentsByRelativeAssetPath(const AZStd::string& relativePath, AZStd::vector<AZStd::string>& dependents) const
     {
         bool found = false;
-        bool succeeded = m_databaseConnection->QuerySourceDependencyByDependsOnSource(
-            relativePath.c_str(),
-            AzToolsFramework::AssetDatabase::SourceFileDependencyEntry::DEP_SourceOrJob,
-            [&](AssetDatabase::SourceFileDependencyEntry& entry)
+
+        auto callback = [&](AssetDatabase::SourceFileDependencyEntry& entry)
         {
             AzToolsFramework::AssetDatabase::SourceDatabaseEntry sourceEntry;
-            m_databaseConnection->QuerySourceBySourceGuid(entry.m_sourceGuid, [&sourceEntry](AzToolsFramework::AssetDatabase::SourceDatabaseEntry& entry)
-            {
-                sourceEntry = entry;
-                return false;
-            });
+            m_databaseConnection->QuerySourceBySourceGuid(
+                entry.m_sourceGuid,
+                [&sourceEntry](AzToolsFramework::AssetDatabase::SourceDatabaseEntry& entry)
+                {
+                    sourceEntry = entry;
+                    return false;
+                });
 
-            if(!sourceEntry.m_sourceName.ends_with(".slice"))
+            if (!sourceEntry.m_sourceName.ends_with(".slice"))
             {
                 // Filter out non-slice files
```

Comment:
could this comment be expanded to explain why the *.slice names need to be filtered out?